In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

## Closed_deals

In [2]:
# carga del dataset y muestreo de 10 registros aleatorios
Closed_deals = pd.read_csv('Datasets/olist_closed_deals_dataset.csv')
Closed_deals.sample(10)

mql_id                         seller_id  \
15   bae38661a27f6228ba38c36e766ed769  6a6b1614baaaf766293c17d8cb8c5a9d   
423  57a80e778f3d755b4f2166e6a60826fd  bea51724befa11664f1d44a92ba8f4c3   
77   496515dec2a1c280f145f6ccb34851fb  6f835fd4be26989b1b064399da346143   
737  a8da3378b39d078f77361620c9481ffb  64c9a1db4e73e19aaafd3286dc448c96   
437  c07c69d716213f5eb297e2721115daca  cf8ca2db83470a73b7cebdd86ea96fb9   
472  5ca05341418f09367540f05484d0b650  fdde1d384dea864917f6b8f2d330986e   
37   a8d2884db49e0769ad4a0b8dce2e143c  d65f74c704563bcb8baa915fbf8cfa1e   
678  8d4af10effffa338cc97d3ef4b2507dd  53088c75f948b48a7ece173ac17f0c12   
6    c3e30ed7ac989117c7e1e719b4ac128f  92d7568ad0c5c76fd7d341b2d46f24d6   
332  d348734a9ee240ebc4c0937a6e755621  f0533bc9cc3111a0b2f8198157a34571   

                               sdr_id                             sr_id  \
15   f42a2bd194f7802ab052a815c8de65b7  495d4e95a8cf8bbf8b432b612a2aa328   
423  de63de0d10a6012430098db33c679b0b  d3d1e91a157ea7f90548eef82f1955e3   
77   09285259593c61296eef10c734121d5b  d3d1e91a157ea7f90548eef82f1955e3   
737  0a0fb2b07d841f84fb6714e35c723075  6565aa9ce3178a5caf6171827af3a9ba   
437  56bf83c4bb35763a51c2baab501b4c67  9ae085775a198122c5586fa830ff7f2b   
472  de63de0d10a6012430098db33c679b0b  4ef15afb4b2723d8f3d81e51ec7afefe   
37   6aa3b86a83d784b05f0e37e26b20860d  de63de0d10a6012430098db33c679b0b   
678  34d40cdaf94010a1d05b0d6212f9e909  068066e24f0c643eb1d089c7dd20cd73   
6    4b339f9567d060bcea4f5136b9f5949e  85fc447d336637ba1df43e793199fbc8   
332  56bf83c4bb35763a51c2baab501b4c67  060c0a26f19f4d66b42e0d8796688490   

                won_date         business_segment        lead_type  \
15   2018-09-10 14:18:01               stationery         industry   
423  2018-04-18 19:01:21                  jewerly  online_beginner   
77   2018-04-27 16:09:25  audio_video_electronics    online_medium   
737  2018-08-07 17:17:28      household_utilities    online_medium   
437  2018-03-21 16:16:52             phone_mobile         industry   
472  2018-01-23 18:30:32                    party          offline   
37   2018-09-27 12:31:01            health_beauty    online_medium   
678  2018-05-09 13:17:12            health_beauty          offline   
6    2018-04-16 18:18:22                computers    online_medium   
332  2018-02-07 18:11:05            health_beauty    online_medium   

    lead_behaviour_profile has_company has_gtin average_stock business_type  \
15                     cat         NaN      NaN         20-50      reseller   
423                    NaN         NaN      NaN           NaN  manufacturer   
77                    wolf         NaN      NaN           NaN      reseller   
737                    cat        True     True          5-20      reseller   
437                  eagle         NaN      NaN           NaN      reseller   
472                    NaN         NaN      NaN           NaN  manufacturer   
37                     cat         NaN    False           NaN      reseller   
678                  eagle         NaN      NaN           NaN      reseller   
6                     wolf         NaN      NaN           NaN      reseller   
332                    NaN         NaN      NaN           NaN      reseller   

     declared_product_catalog_size  declared_monthly_revenue  
15                          2000.0                  100000.0  
423                            NaN                       0.0  
77                             NaN                       0.0  
737                            NaN                       0.0  
437                            NaN                       0.0  
472                            NaN                       0.0  
37                            15.0                   20000.0  
678                            NaN                       0.0  
6                              NaN                       0.0  
332                            NaN                       0.0

In [3]:
# mombres de columnas
Closed_deals.columns

Index(['mql_id', 'seller_id', 'sdr_id', 'sr_id', 'won_date',
       'business_segment', 'lead_type', 'lead_behaviour_profile',
       'has_company', 'has_gtin', 'average_stock', 'business_type',
       'declared_product_catalog_size', 'declared_monthly_revenue'],
      dtype='object')

In [4]:
# tamaño del dataset
Closed_deals.shape

(842, 14)

In [5]:
# tipo de datos, no nulos y cantidad de registros para cada columna
Closed_deals.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 842 entries, 0 to 841
Data columns (total 14 columns):
 #   Column                         Non-Null Count  Dtype  
---  ------                         --------------  -----  
 0   mql_id                         842 non-null    object 
 1   seller_id                      842 non-null    object 
 2   sdr_id                         842 non-null    object 
 3   sr_id                          842 non-null    object 
 4   won_date                       842 non-null    object 
 5   business_segment               841 non-null    object 
 6   lead_type                      836 non-null    object 
 7   lead_behaviour_profile         665 non-null    object 
 8   has_company                    63 non-null     object 
 9   has_gtin                       64 non-null     object 
 10  average_stock                  66 non-null     object 
 11  business_type                  832 non-null    object 
 12  declared_product_catalog_size  69 non-null     flo

In [6]:
# descripcion de las columnas numéricas
Closed_deals.describe()

declared_product_catalog_size  declared_monthly_revenue
count                      69.000000              8.420000e+02
mean                      233.028986              7.337768e+04
std                       352.380558              1.744799e+06
min                         1.000000              0.000000e+00
25%                        30.000000              0.000000e+00
50%                       100.000000              0.000000e+00
75%                       300.000000              0.000000e+00
max                      2000.000000              5.000000e+07

In [7]:
# como se comportan las variables categóricas-- count=cantidad, unique=registros unicos, top=mas veces registrado, freq=frecuencia de los registros con mas aparariociones
Closed_deals.describe(include=['O'])

mql_id                         seller_id  \
count                                842                               842   
unique                               842                               842   
top     5420aad7fec3549a85876ba1c529bd84  2c43fb513632d29b3b58df74816f1b06   
freq                                   1                                 1   

                                  sdr_id                             sr_id  \
count                                842                               842   
unique                                32                                22   
top     4b339f9567d060bcea4f5136b9f5949e  4ef15afb4b2723d8f3d81e51ec7afefe   
freq                                 140                               133   

                   won_date business_segment      lead_type  \
count                   842              841            836   
unique                  824               33              8   
top     2018-05-04 03:00:00       home_decor  online_medium   
freq                      6              105            332   

       lead_behaviour_profile has_company has_gtin average_stock business_type  
count                     665          63       64            66           832  
unique                      9           2        2             6             3  
top                       cat        True     True          5-20      reseller  
freq                      407          58       54            22           587

In [8]:
# cuantos datos nulos por columna se encuentran
Closed_deals.isna().sum()

mql_id                             0
seller_id                          0
sdr_id                             0
sr_id                              0
won_date                           0
business_segment                   1
lead_type                          6
lead_behaviour_profile           177
has_company                      779
has_gtin                         778
average_stock                    776
business_type                     10
declared_product_catalog_size    773
declared_monthly_revenue           0
dtype: int64

- De un total de 842 registros las columnas has_company, has_gtin, average_stock y declared_product_catalog_size tiene mas del 90 % de sus datos faltantes

## Customers

In [9]:
# carga del dataset y muestreo de 10 registros aleatorios
Customers = pd.read_csv('Datasets/olist_customers_dataset.csv')
Customers.sample(10)

customer_id                customer_unique_id  \
72017  a459383f702f0417d0d65fff9945aac1  9d2ce111f0f71e84ff66d19cca09dd5b   
48004  65d9c386ee0eb43e165aded25bff79d7  1f3134e247f3bb66b9aaecd65382c587   
98417  fff906ecb75de5809be384e0f8d65e45  948250fc2b818dd22924a8000d69700d   
94944  976f152ef42f30a6e4bca8966a8b4099  f5404ba53c69cca35b1e7f696dcadeb1   
86315  6b9c443a3f61f5c12ba58b9c84e1e22b  2c773508172a15be27bceb07410577a7   
77614  ca8e595c4c2e4a55fa75a41d5bf08382  8c108bbb0e50260292b20cc76a647aaf   
46589  6cd80ca371f4be9a7bd93aefcf6dfcb7  8ad26fa5294b5843055e1d1aec0eb1d8   
41635  7eba295b7f6a16dcf3e9f75510d94898  0049dadc30da6a277c1e9d371df5a66f   
57557  4b0f1543a7439e9abd2078fbefd3f8c3  727260949d242157d3db0083e782665b   
37437  70ab5e994ea383619a19687837eedbbc  9fdda3b176806e4447ead693e25ee82a   

       customer_zip_code_prefix         customer_city customer_state  
72017                     27281         volta redonda             RJ  
48004                     37408  sao thome das letras             MG  
98417                     35170    coronel fabriciano             MG  
94944                      9070           santo andre             SP  
86315                     60055             fortaleza             CE  
77614                     13484               limeira             SP  
46589                     75860          quirinopolis             GO  
41635                     70680              brasilia             DF  
57557                     41630              salvador             BA  
37437                      4474             sao paulo             SP

In [10]:
# nombres de columnas
Customers.columns

Index(['customer_id', 'customer_unique_id', 'customer_zip_code_prefix',
       'customer_city', 'customer_state'],
      dtype='object')

In [11]:
# tamaño del dataset
Customers.shape

(99441, 5)

In [12]:
# tipo de datos, no nulos y cantidad de registros para cada columna
Customers.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 99441 entries, 0 to 99440
Data columns (total 5 columns):
 #   Column                    Non-Null Count  Dtype 
---  ------                    --------------  ----- 
 0   customer_id               99441 non-null  object
 1   customer_unique_id        99441 non-null  object
 2   customer_zip_code_prefix  99441 non-null  int64 
 3   customer_city             99441 non-null  object
 4   customer_state            99441 non-null  object
dtypes: int64(1), object(4)
memory usage: 3.8+ MB


In [13]:
# descripcion de las columnas numéricas 
Customers.describe()

customer_zip_code_prefix
count              99441.000000
mean               35137.474583
std                29797.938996
min                 1003.000000
25%                11347.000000
50%                24416.000000
75%                58900.000000
max                99990.000000

In [14]:
# como se comportan las variables categóricas-- count=cantidad, unique=registros unicos, top=mas veces registrado, freq=frecuencia de los registros con mas aparariociones
Customers.describe(include=['O'])

customer_id                customer_unique_id  \
count                              99441                             99441   
unique                             99441                             96096   
top     06b8999e2fba1a1fbc88172c00ba8bc7  8d50f5eadf50201ccdcedfb9e2ac8455   
freq                                   1                                17   

       customer_city customer_state  
count          99441          99441  
unique          4119             27  
top        sao paulo             SP  
freq           15540          41746

- analizando Customers faltan datos de customers_unique_id que no hay referencia de como fueron creados esos codigos

In [15]:
# cuantos datos nulos por solumna se encuentran
Customers.isna().sum()

customer_id                 0
customer_unique_id          0
customer_zip_code_prefix    0
customer_city               0
customer_state              0
dtype: int64

## Geolocation

In [16]:
# carga del dataset y muestreo de 10 registros aleatorios
Geolocation = pd.read_csv('Datasets/olist_geolocation_dataset.csv')
Geolocation.sample(10)

geolocation_zip_code_prefix  geolocation_lat  geolocation_lng  \
556905                        30750       -19.899538       -43.976886   
715533                        53431        -7.903548       -34.836745   
156219                         6390       -23.540203       -46.847964   
400487                        19470       -21.766636       -52.112499   
529898                        29160       -20.230771       -40.274145   
332593                        14169       -21.141628       -47.977734   
537798                        29560       -20.770679       -41.673460   
977760                        96170       -31.362460       -51.971037   
232791                         9581       -23.637653       -46.581177   
690555                        45400       -13.371195       -39.075197   

           geolocation_city geolocation_state  
556905       belo horizonte                MG  
715533             paulista                PE  
156219          carapicuiba                SP  
400487  presidente epitacio                SP  
529898                serra                ES  
332593          sertãozinho                SP  
537798               guacui                ES  
977760  são lourenço do sul                RS  
232791   sao caetano do sul                SP  
690555              valenca                BA

In [17]:
# nombres de columnas
Geolocation.columns

Index(['geolocation_zip_code_prefix', 'geolocation_lat', 'geolocation_lng',
       'geolocation_city', 'geolocation_state'],
      dtype='object')

In [18]:
# tamaño del dataset
Geolocation.shape

(1000163, 5)

In [19]:
# tipo de datos, no nulos y cantidad de registros para cada columna
Geolocation.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000163 entries, 0 to 1000162
Data columns (total 5 columns):
 #   Column                       Non-Null Count    Dtype  
---  ------                       --------------    -----  
 0   geolocation_zip_code_prefix  1000163 non-null  int64  
 1   geolocation_lat              1000163 non-null  float64
 2   geolocation_lng              1000163 non-null  float64
 3   geolocation_city             1000163 non-null  object 
 4   geolocation_state            1000163 non-null  object 
dtypes: float64(2), int64(1), object(2)
memory usage: 38.2+ MB


In [20]:
# descripcion de las columnas numéricas
Geolocation.describe()

geolocation_zip_code_prefix  geolocation_lat  geolocation_lng
count                 1.000163e+06     1.000163e+06     1.000163e+06
mean                  3.657417e+04    -2.117615e+01    -4.639054e+01
std                   3.054934e+04     5.715866e+00     4.269748e+00
min                   1.001000e+03    -3.660537e+01    -1.014668e+02
25%                   1.107500e+04    -2.360355e+01    -4.857317e+01
50%                   2.653000e+04    -2.291938e+01    -4.663788e+01
75%                   6.350400e+04    -1.997962e+01    -4.376771e+01
max                   9.999000e+04     4.506593e+01     1.211054e+02

In [21]:
# como se comportan las variables categóricas-- count=cantidad, unique=registros unicos, top=mas veces registrado, freq=frecuencia de los registros con mas aparariociones
Geolocation.describe(include=['O'])

geolocation_city geolocation_state
count           1000163           1000163
unique             8011                27
top           sao paulo                SP
freq             135800            404268

In [22]:
# cuantos datos nulos por columna se encuentran
Geolocation.isna().sum()

geolocation_zip_code_prefix    0
geolocation_lat                0
geolocation_lng                0
geolocation_city               0
geolocation_state              0
dtype: int64

## Marketing

In [23]:
# carga del dataset y muestreo de 10 registros aleatorios
Marketing = pd.read_csv('Datasets/olist_marketing_qualified_leads_dataset.csv')
Marketing.sample(10)

mql_id first_contact_date  \
5888  b9b4d4f5b575539161c184faf8e53d5f         2018-05-14   
729   437d2c4de406d5e46cb45e9802811f31         2018-01-13   
7621  eb968139ca829c6e4192e49baa4a0578         2018-03-25   
6174  e8644ee27d873e0bb207499b0279b8e8         2017-11-28   
1669  1d5fe42fe9d32afaa94d6b020be8384c         2018-01-06   
6599  a7ad2cab998ae504104b31b75758440e         2018-01-26   
1558  1632af4762431469541ac66e2d6f4b45         2018-02-09   
1564  eeb3b040d8d25cd782183ae757fdbfd0         2018-02-28   
3157  d348627a08fabe278a8909ed03474447         2018-05-10   
3386  9fd3ac60c2e7a7b4b5984fb48bda8492         2017-09-20   

                       landing_page_id          origin  
5888  b76ef37428e6799c421989521c0e5077         unknown  
729   d83b0d0e48c8447d1d5507a44027a955     paid_search  
7621  58326e62183c14b0c03085c33b9fdc44          social  
6174  f98c2bd96ac4a5d9476734e790c91384  organic_search  
1669  489a340e7e59e0bcbec67b4a19c1b229     paid_search  
6599  da14b7e2eb0d1eecf65abbd05a31c87d  organic_search  
1558  f98c2bd96ac4a5d9476734e790c91384  organic_search  
1564  40dec9f3d5259a3d2dbcdab2114fae47     paid_search  
3157  d75a2371281f42d607be43b08bd74905  direct_traffic  
3386  30077c17f2ec5010a82e37ad8925b95f     paid_search

In [24]:
# mombres de columnas
Marketing.columns

Index(['mql_id', 'first_contact_date', 'landing_page_id', 'origin'], dtype='object')

In [25]:
# tamaño del dataset
Marketing.shape

(8000, 4)

In [26]:
# tipo de datos, no nulos y cantidad de registros para cada columna
Marketing.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8000 entries, 0 to 7999
Data columns (total 4 columns):
 #   Column              Non-Null Count  Dtype 
---  ------              --------------  ----- 
 0   mql_id              8000 non-null   object
 1   first_contact_date  8000 non-null   object
 2   landing_page_id     8000 non-null   object
 3   origin              7940 non-null   object
dtypes: object(4)
memory usage: 250.1+ KB


In [27]:
# descripcion de las columnas numéricas
Marketing.describe()

mql_id first_contact_date  \
count                               8000               8000   
unique                              8000                336   
top     dac32acd4db4c29c230538b72f8dd87d         2018-05-02   
freq                                   1                 93   

                         landing_page_id          origin  
count                               8000            7940  
unique                               495              10  
top     b76ef37428e6799c421989521c0e5077  organic_search  
freq                                 912            2296

In [28]:
# como se comportan las variables categóricas-- count=cantidad, unique=registros unicos, top=mas veces registrado, freq=frecuencia de los registros con mas aparariociones
Marketing.describe(include=['O'])

mql_id first_contact_date  \
count                               8000               8000   
unique                              8000                336   
top     dac32acd4db4c29c230538b72f8dd87d         2018-05-02   
freq                                   1                 93   

                         landing_page_id          origin  
count                               8000            7940  
unique                               495              10  
top     b76ef37428e6799c421989521c0e5077  organic_search  
freq                                 912            2296

In [29]:
# cuantos datos nulos por columna se encuentran
Marketing.isna().sum()

mql_id                 0
first_contact_date     0
landing_page_id        0
origin                60
dtype: int64

## Order_items

In [30]:
# carga del dataset y muestreo de 10 registros aleatorios
Order_items = pd.read_csv('Datasets/olist_order_items_dataset.csv')
Order_items.sample(10)

order_id  order_item_id  \
1171   02b8fcdd1bd75d7d5861f407dde29768              1   
84465  bfe2018d49712c5c36ce0ce9307249a1              1   
90402  cd4d00f4cdfc2365a398924baf0defa7              1   
10890  18cc97889312466cc90150237a3e2603              1   
92874  d29601cbe18ae8190a14e560a413b890              1   
2308   05408ed33df5e0bbf70d578dbdca2f29              1   
99952  e2add76d15b3e6cd98abe9947f421800              1   
92800  d2741de337666d1bb01913f415e6643f              1   
13456  1e90a410c4be85320d64e5a0cbed9291              1   
80304  b6b7c5714dc60f5699ec29abcb31cf50              1   

                             product_id                         seller_id  \
1171   8817f0cda3d5b42e5f949ee3be9dc80c  3c7c4a49ec3c6550809089c6a2ca9370   
84465  d84e43351df6812fc8bf1c70405fb42e  18a349e75d307f4b4cc646a691ed4216   
90402  73bcea88583002fe12b7399af4ae60da  4869f7a5dfa277a7dca6462dcf3b52b2   
10890  fe61253e50ad6ee9f1d180da22ce5eef  39d54ff918774174706fb065d7f9dc07   
92874  d061d81f407dcbadbe325a56a93d80a3  43f8c9950d11ecd03a0304a49e010da6   
2308   bcfef6d5629403cedb2c8b855940337c  b4ffb71f0cb1b1c3d63fad021ecf93e1   
99952  d81d3da04eaeaf9608d87763815d7885  73da1880b39c71de2b262bc7720c1494   
92800  5739177f0e1a8c060da43932f5a447a8  5bffbafbb22daf6d3bfc216a46db8708   
13456  c781fcf450227132b560b10bbd2ed26b  827f8f69dfa529c561901c4f2e0f332f   
80304  b657163c168e6b574fdab182100742b1  42ef3192a9ff87a22d1867b74b3ee205   

       shipping_limit_date   price  freight_value  
1171   2018-05-18 11:19:08   13.99          22.85  
84465  2018-02-26 13:30:29  182.32          13.62  
90402  2018-03-27 09:48:17  134.90          18.82  
10890  2017-12-14 15:31:37   99.00          18.26  
92874  2017-10-26 21:14:50  199.99          15.15  
2308   2018-08-02 16:45:16   34.50           7.50  
99952  2017-08-03 08:43:40  199.00          19.28  
92800  2017-03-02 14:50:18   18.99          17.78  
13456  2017-06-20 20:05:10   29.90          14.10  
80304  2018-04-05 23:27:21   14.90           7.39

In [31]:
# mombres de columnas
Order_items.columns

Index(['order_id', 'order_item_id', 'product_id', 'seller_id',
       'shipping_limit_date', 'price', 'freight_value'],
      dtype='object')

In [32]:
# tamaño del dataset
Order_items.shape

(112650, 7)

In [33]:
# tipo de datos, no nulos y cantidad de registros para cada columna
Order_items.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 112650 entries, 0 to 112649
Data columns (total 7 columns):
 #   Column               Non-Null Count   Dtype  
---  ------               --------------   -----  
 0   order_id             112650 non-null  object 
 1   order_item_id        112650 non-null  int64  
 2   product_id           112650 non-null  object 
 3   seller_id            112650 non-null  object 
 4   shipping_limit_date  112650 non-null  object 
 5   price                112650 non-null  float64
 6   freight_value        112650 non-null  float64
dtypes: float64(2), int64(1), object(4)
memory usage: 6.0+ MB


In [34]:
# descripcion de las columnas numéricas
Order_items.describe()

order_item_id          price  freight_value
count  112650.000000  112650.000000  112650.000000
mean        1.197834     120.653739      19.990320
std         0.705124     183.633928      15.806405
min         1.000000       0.850000       0.000000
25%         1.000000      39.900000      13.080000
50%         1.000000      74.990000      16.260000
75%         1.000000     134.900000      21.150000
max        21.000000    6735.000000     409.680000

In [35]:
# como se comportan las variables categóricas-- count=cantidad, unique=registros unicos, top=mas veces registrado, freq=frecuencia de los registros con mas aparariociones
Order_items.describe(include=['O'])

order_id                        product_id  \
count                             112650                            112650   
unique                             98666                             32951   
top     8272b63d03f5f79c56e9e4120aec44ef  aca2eb7d00ea1a7b8ebd4e68314663af   
freq                                  21                               527   

                               seller_id  shipping_limit_date  
count                             112650               112650  
unique                              3095                93318  
top     6560211a19b47992c3666cc44a7e94c0  2017-07-21 18:25:23  
freq                                2033                   21

In [36]:
# cuantos datos nulos por columna se encuentran
Order_items.isna().sum()

order_id               0
order_item_id          0
product_id             0
seller_id              0
shipping_limit_date    0
price                  0
freight_value          0
dtype: int64

## Order_payments

In [37]:
# carga del dataset y muestreo de 10 registros aleatorios
Order_payments = pd.read_csv('Datasets/olist_order_payments_dataset.csv')
Order_payments.sample(10)

order_id  payment_sequential payment_type  \
22906  d3b8750f21e85ef7a244c8a0cb8ae7b6                   1       boleto   
84424  ffa466d3e55f98477b7e0da764df3c6c                   1  credit_card   
37533  3f1ba9515b44ee835d8fcb64238f0b27                   1  credit_card   
23931  dc8ffb2856b34d44541032dff3d0eb5c                   1   debit_card   
32909  610550e46b3a0a7f45723db57f2170f3                   1  credit_card   
87389  bc7378e3329da5c228531efaaabb18ae                   1  credit_card   
4958   d4c0ebae1ec07474b7d092a62b469f6d                   1  credit_card   
87696  d8c3814df9c4a328159b280e60a59486                   1  credit_card   
8568   54e9851195e7b44a91adc2564e4d082c                   3      voucher   
97262  1b489d892646547f7e27765f3438d8b6                   1  credit_card   

       payment_installments  payment_value  
22906                     1          85.14  
84424                     1          93.36  
37533                     8         596.86  
23931                     1          39.60  
32909                     7          76.65  
87389                    10         174.21  
4958                      1          11.91  
87696                     4        1728.81  
8568                      1         135.66  
97262                     1          46.42

In [38]:
# mombres de columnas
Order_payments.columns

Index(['order_id', 'payment_sequential', 'payment_type',
       'payment_installments', 'payment_value'],
      dtype='object')

In [39]:
# tamaño del dataset
Order_payments.shape

(103886, 5)

In [40]:
# tipo de datos, no nulos y cantidad de registros para cada columna
Order_payments.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 103886 entries, 0 to 103885
Data columns (total 5 columns):
 #   Column                Non-Null Count   Dtype  
---  ------                --------------   -----  
 0   order_id              103886 non-null  object 
 1   payment_sequential    103886 non-null  int64  
 2   payment_type          103886 non-null  object 
 3   payment_installments  103886 non-null  int64  
 4   payment_value         103886 non-null  float64
dtypes: float64(1), int64(2), object(2)
memory usage: 4.0+ MB


In [41]:
# descripcion de las columnas numéricas
Order_payments.describe()

payment_sequential  payment_installments  payment_value
count       103886.000000         103886.000000  103886.000000
mean             1.092679              2.853349     154.100380
std              0.706584              2.687051     217.494064
min              1.000000              0.000000       0.000000
25%              1.000000              1.000000      56.790000
50%              1.000000              1.000000     100.000000
75%              1.000000              4.000000     171.837500
max             29.000000             24.000000   13664.080000

In [42]:
# como se comportan las variables categóricas-- count=cantidad, unique=registros unicos, top=mas veces registrado, freq=frecuencia de los registros con mas aparariociones
Order_payments.describe(include=['O'])

order_id payment_type
count                             103886       103886
unique                             99440            5
top     fa65dad1b0e818e3ccc5cb0e39231352  credit_card
freq                                  29        76795

In [43]:
# cuantos datos nulos por columna se encuentran
Order_payments.isna().sum()

order_id                0
payment_sequential      0
payment_type            0
payment_installments    0
payment_value           0
dtype: int64

## Order_reviews

In [44]:
# carga del dataset y muestreo de 10 registros aleatorios
Order_reviews = pd.read_csv('Datasets/olist_order_reviews_dataset.csv')
Order_reviews.sample(10)

review_id                          order_id  \
33035  91de7da06aa7772a56afe4f7f88b9dc5  77af320be1815e066c2ac8510e9402b6   
49116  b240e60b7c2d944ce479ceb942ccea07  d40956f82498404a78bdefbed0d005b1   
63897  7ea08db468643c2305022ee1b8f842d3  785e5848b43e60ee92f14e0e20dca105   
33350  9c4ce888a4317d4dab21e9e5cf172764  d1d889b705a1d015ae3040fe1cb24be7   
63926  923a355eb062b71fe0bf5e603739dadf  0b9e555377748cc93362b079f2653a42   
73644  827a4677b7620e6de532710933cdc217  fddd8f5e5947098f75c2967d7cb2c3de   
73129  c65bb18311f3d6c8debeb47ba3202e4e  5ddffd82198ca5fd512883316c2c865b   
87095  0ad009979f236d92cdb32fb77ebcd981  e44260bfec2df487a4c92f6fabfe11dd   
96880  f0b329c38f9a842f40b22fe379c5e621  ac0716aa90541fcb9f100e8a7b38e2bf   
97695  e00eb976a9c4d67f1eb13e2c7e42ab7a  5f37c7c9cfd25572ee87b69b77f27b8c   

       review_score review_comment_title  \
33035             5                  NaN   
49116             5            Muito boa   
63897             5                  NaN   
33350             5            recomendo   
63926             5                  NaN   
73644             4                  NaN   
73129             3                  NaN   
87095             5                  NaN   
96880             5                  NaN   
97695             5                  NaN   

                                  review_comment_message review_creation_date  \
33035                                              Otimo  2018-05-12 00:00:00   
49116        Recomendo e se precisar voltarei à comprar.  2018-05-06 00:00:00   
63897                                          perfeito!  2018-03-29 00:00:00   
33350                                                NaN  2018-07-13 00:00:00   
63926                                                NaN  2017-12-22 00:00:00   
73644  O produto chegou bem antes do prazo estipulado...  2017-07-27 00:00:00   
73129                                                NaN  2018-03-02 00:00:00   
87095                                                NaN  2018-01-04 00:00:00   
96880                                          recomendo  2017-10-03 00:00:00   
97695                                                NaN  2018-03-20 00:00:00   

      review_answer_timestamp  
33035     2018-05-14 11:42:50  
49116     2018-05-06 21:46:45  
63897     2018-04-02 12:38:33  
33350     2018-07-14 01:56:05  
63926     2017-12-23 22:04:27  
73644     2017-07-27 10:31:13  
73129     2018-03-03 00:14:27  
87095     2018-01-07 18:00:21  
96880     2017-10-06 18:20:37  
97695     2018-03-22 22:27:36

In [45]:
# mombres de columnas
Order_reviews.columns

Index(['review_id', 'order_id', 'review_score', 'review_comment_title',
       'review_comment_message', 'review_creation_date',
       'review_answer_timestamp'],
      dtype='object')

In [46]:
# tamaño del dataset
Order_reviews.shape

(99224, 7)

In [47]:
# tipo de datos, no nulos y cantidad de registros para cada columna
Order_reviews.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 99224 entries, 0 to 99223
Data columns (total 7 columns):
 #   Column                   Non-Null Count  Dtype 
---  ------                   --------------  ----- 
 0   review_id                99224 non-null  object
 1   order_id                 99224 non-null  object
 2   review_score             99224 non-null  int64 
 3   review_comment_title     11568 non-null  object
 4   review_comment_message   40977 non-null  object
 5   review_creation_date     99224 non-null  object
 6   review_answer_timestamp  99224 non-null  object
dtypes: int64(1), object(6)
memory usage: 5.3+ MB


In [48]:
# descripcion de las columnas numéricas
Order_reviews.describe()

review_score
count  99224.000000
mean       4.086421
std        1.347579
min        1.000000
25%        4.000000
50%        5.000000
75%        5.000000
max        5.000000

In [49]:
# como se comportan las variables categóricas-- count=cantidad, unique=registros unicos, top=mas veces registrado, freq=frecuencia de los registros con mas aparariociones
Order_reviews.describe(include=['O'])

review_id                          order_id  \
count                              99224                             99224   
unique                             98410                             98673   
top     7b606b0d57b078384f0b58eac1d41d78  c88b1d1b157a9999ce368f218a407141   
freq                                   3                                 3   

       review_comment_title review_comment_message review_creation_date  \
count                 11568                  40977                99224   
unique                 4527                  36159                  636   
top               Recomendo              Muito bom  2017-12-19 00:00:00   
freq                    423                    230                  463   

       review_answer_timestamp  
count                    99224  
unique                   98248  
top        2017-06-15 23:21:05  
freq                         4

In [50]:
# cuantos datos nulos por columna se encuentran
Order_reviews.isna().sum()

review_id                      0
order_id                       0
review_score                   0
review_comment_title       87656
review_comment_message     58247
review_creation_date           0
review_answer_timestamp        0
dtype: int64

## Orders

In [51]:
# carga del dataset y muestreo de 10 registros aleatorios
Orders = pd.read_csv('Datasets/olist_orders_dataset.csv')
Orders.sample(10)

order_id                       customer_id  \
13721  669eb4808ac7fe964a716a368943ddfa  0c562074c5f964adc5bcfa39d647a834   
20953  e8a2fe3e5e58d8ed9080188f35c108ef  9dd8245cbdc86cf19b4e02f11c923849   
4055   fb4c654f040642300a388f17ee02be7f  d9e3358d714896d16cbc712a46559861   
85823  97e080a21be085e3e36d5f7c0f8f0b43  92b28c41e27f5ad13c88675d208d3d3a   
84828  d62354cddc18720cf8a24b521570a8f7  97170ff49eb12980f83d493178bc8e66   
38607  68a08e0bf6b9dd35f0bb977acd0f1358  09436a6184161df39d8528de4e13b7dc   
8066   29acb175de2a4da2b456200329d0f5ca  830fada6ce508f4799c9c927fed6acd0   
12539  f340f140ae72544b6df944a32ccba7c0  18207bfa3ee93516926cc203d7558453   
38842  f354a10235b5d253194c5285396abad1  8d37602184dc432a7ca227475ac81036   
72674  e1df61c61699c690427a33fc1b6b540c  4c44865a36098a8b24dba15c90f01511   

      order_status order_purchase_timestamp    order_approved_at  \
13721    delivered      2018-04-10 18:32:57  2018-04-12 04:55:24   
20953    delivered      2017-06-13 11:13:23  2017-06-13 12:23:34   
4055     delivered      2017-11-25 16:49:00  2017-11-25 16:57:26   
85823    delivered      2017-11-01 14:36:07  2017-11-01 14:50:13   
84828    delivered      2018-08-17 22:52:18  2018-08-17 23:09:37   
38607    delivered      2018-03-15 06:35:49  2018-03-15 06:50:20   
8066     delivered      2018-03-31 18:26:48  2018-03-31 18:47:21   
12539    delivered      2017-08-16 15:51:03  2017-08-16 16:55:09   
38842    delivered      2017-05-12 13:46:05  2017-05-16 03:35:38   
72674    delivered      2017-02-09 11:24:15  2017-02-09 11:35:21   

      order_delivered_carrier_date order_delivered_customer_date  \
13721          2018-04-16 21:04:15           2018-04-30 16:28:31   
20953          2017-06-19 13:57:55           2017-06-19 18:07:51   
4055           2017-11-27 20:35:53           2017-12-04 22:52:24   
85823          2017-11-09 15:26:48           2017-11-14 17:39:31   
84828          2018-08-20 09:25:00           2018-08-21 15:50:58   
38607          2018-03-16 19:46:27           2018-04-13 19:44:28   
8066           2018-04-02 16:52:12           2018-04-09 18:47:55   
12539          2017-08-17 19:29:37           2017-08-18 18:42:35   
38842          2017-05-16 13:38:34           2017-05-23 14:37:30   
72674          2017-02-10 08:34:09           2017-02-20 11:51:52   

      order_estimated_delivery_date  
13721           2018-05-15 00:00:00  
20953           2017-06-27 00:00:00  
4055            2017-12-18 00:00:00  
85823           2017-11-28 00:00:00  
84828           2018-08-23 00:00:00  
38607           2018-04-09 00:00:00  
8066            2018-05-11 00:00:00  
12539           2017-09-01 00:00:00  
38842           2017-06-08 00:00:00  
72674           2017-03-16 00:00:00

In [52]:
# mombres de columnas
Orders.columns

Index(['order_id', 'customer_id', 'order_status', 'order_purchase_timestamp',
       'order_approved_at', 'order_delivered_carrier_date',
       'order_delivered_customer_date', 'order_estimated_delivery_date'],
      dtype='object')

In [53]:
# tamaño del dataset
Orders.shape

(99441, 8)

In [54]:
# tipo de datos, no nulos y cantidad de registros para cada columna
Orders.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 99441 entries, 0 to 99440
Data columns (total 8 columns):
 #   Column                         Non-Null Count  Dtype 
---  ------                         --------------  ----- 
 0   order_id                       99441 non-null  object
 1   customer_id                    99441 non-null  object
 2   order_status                   99441 non-null  object
 3   order_purchase_timestamp       99441 non-null  object
 4   order_approved_at              99281 non-null  object
 5   order_delivered_carrier_date   97658 non-null  object
 6   order_delivered_customer_date  96476 non-null  object
 7   order_estimated_delivery_date  99441 non-null  object
dtypes: object(8)
memory usage: 6.1+ MB


In [55]:
# descripcion de las columnas numéricas
Orders.describe()

order_id                       customer_id  \
count                              99441                             99441   
unique                             99441                             99441   
top     e481f51cbdc54678b7cc49136f2d6af7  9ef432eb6251297304e76186b10a928d   
freq                                   1                                 1   

       order_status order_purchase_timestamp    order_approved_at  \
count         99441                    99441                99281   
unique            8                    98875                90733   
top       delivered      2018-04-11 10:48:14  2018-02-27 04:31:10   
freq          96478                        3                    9   

       order_delivered_carrier_date order_delivered_customer_date  \
count                         97658                         96476   
unique                        81018                         95664   
top             2018-05-09 15:48:00           2018-05-08 23:38:46   
freq                             47                             3   

       order_estimated_delivery_date  
count                          99441  
unique                           459  
top              2017-12-20 00:00:00  
freq                             522

In [56]:
# como se comportan las variables categóricas-- count=cantidad, unique=registros unicos, top=mas veces registrado, freq=frecuencia de los registros con mas aparariociones
Orders.describe(include=['O'])

order_id                       customer_id  \
count                              99441                             99441   
unique                             99441                             99441   
top     e481f51cbdc54678b7cc49136f2d6af7  9ef432eb6251297304e76186b10a928d   
freq                                   1                                 1   

       order_status order_purchase_timestamp    order_approved_at  \
count         99441                    99441                99281   
unique            8                    98875                90733   
top       delivered      2018-04-11 10:48:14  2018-02-27 04:31:10   
freq          96478                        3                    9   

       order_delivered_carrier_date order_delivered_customer_date  \
count                         97658                         96476   
unique                        81018                         95664   
top             2018-05-09 15:48:00           2018-05-08 23:38:46   
freq                             47                             3   

       order_estimated_delivery_date  
count                          99441  
unique                           459  
top              2017-12-20 00:00:00  
freq                             522

In [57]:
# cuantos datos nulos por columna se encuentran
Orders.isna().sum()

order_id                            0
customer_id                         0
order_status                        0
order_purchase_timestamp            0
order_approved_at                 160
order_delivered_carrier_date     1783
order_delivered_customer_date    2965
order_estimated_delivery_date       0
dtype: int64

## Products

In [58]:
# carga del dataset y muestreo de 10 registros aleatorios
Products = pd.read_csv('Datasets/olist_products_dataset.csv')
Products.sample(10)

product_id  product_category_name  \
28146  3bf28879e1992bc098404cbf06003647              telefonia   
14907  128d1c88b601731fe8ed4044b1ae7a6a        cama_mesa_banho   
31064  af893b2efa430418d774a852e7870601  utilidades_domesticas   
13369  7ac92e616265ecb6de1477bd80277509             automotivo   
22216  74f6a88474993377506706e383f98302              cine_foto   
19815  2c36bd802c645850497100e08f9dfc80           beleza_saude   
15409  805fb04b1eee16776d3ff45a40a30044      moveis_escritorio   
13310  881218c6167982e1dd94cb98a085cbb9       eletrodomesticos   
2782   9afc7bd1c6286201ccf4bf74ffc9ce6b  utilidades_domesticas   
9717   e0bb3159111d86828b5caef8725ff055  utilidades_domesticas   

       product_name_lenght  product_description_lenght  product_photos_qty  \
28146                 55.0                      1323.0                 2.0   
14907                 50.0                       120.0                 1.0   
31064                 59.0                       566.0                 3.0   
13369                 39.0                       212.0                 2.0   
22216                 55.0                       876.0                 1.0   
19815                 55.0                       931.0                 1.0   
15409                 58.0                      1295.0                 2.0   
13310                 48.0                        55.0                 1.0   
2782                  27.0                        59.0                 2.0   
9717                  63.0                       103.0                 1.0   

       product_weight_g  product_length_cm  product_height_cm  \
28146              50.0               18.0                6.0   
14907             650.0               34.0               18.0   
31064            1550.0               29.0               17.0   
13369            1000.0               19.0               19.0   
22216            1900.0               30.0               13.0   
19815            3598.0               41.0               18.0   
15409            3875.0               35.0               52.0   
13310             550.0               36.0               15.0   
2782              250.0               23.0               13.0   
9717             2500.0               30.0                6.0   

       product_width_cm  
28146              11.0  
14907              44.0  
31064              23.0  
13369              14.0  
22216              30.0  
19815              51.0  
15409              34.0  
13310              25.0  
2782               21.0  
9717               30.0

In [59]:
# mombres de columnas
Products.columns

Index(['product_id', 'product_category_name', 'product_name_lenght',
       'product_description_lenght', 'product_photos_qty', 'product_weight_g',
       'product_length_cm', 'product_height_cm', 'product_width_cm'],
      dtype='object')

In [60]:
# tamaño del dataset
Products.shape

(32951, 9)

In [61]:
# tipo de datos, no nulos y cantidad de registros para cada columna
Products.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 32951 entries, 0 to 32950
Data columns (total 9 columns):
 #   Column                      Non-Null Count  Dtype  
---  ------                      --------------  -----  
 0   product_id                  32951 non-null  object 
 1   product_category_name       32341 non-null  object 
 2   product_name_lenght         32341 non-null  float64
 3   product_description_lenght  32341 non-null  float64
 4   product_photos_qty          32341 non-null  float64
 5   product_weight_g            32949 non-null  float64
 6   product_length_cm           32949 non-null  float64
 7   product_height_cm           32949 non-null  float64
 8   product_width_cm            32949 non-null  float64
dtypes: float64(7), object(2)
memory usage: 2.3+ MB


In [62]:
# descripcion de las columnas numéricas
Products.describe()

product_name_lenght  product_description_lenght  product_photos_qty  \
count         32341.000000                32341.000000        32341.000000   
mean             48.476949                  771.495285            2.188986   
std              10.245741                  635.115225            1.736766   
min               5.000000                    4.000000            1.000000   
25%              42.000000                  339.000000            1.000000   
50%              51.000000                  595.000000            1.000000   
75%              57.000000                  972.000000            3.000000   
max              76.000000                 3992.000000           20.000000   

       product_weight_g  product_length_cm  product_height_cm  \
count      32949.000000       32949.000000       32949.000000   
mean        2276.472488          30.815078          16.937661   
std         4282.038731          16.914458          13.637554   
min            0.000000           7.000000           2.000000   
25%          300.000000          18.000000           8.000000   
50%          700.000000          25.000000          13.000000   
75%         1900.000000          38.000000          21.000000   
max        40425.000000         105.000000         105.000000   

       product_width_cm  
count      32949.000000  
mean          23.196728  
std           12.079047  
min            6.000000  
25%           15.000000  
50%           20.000000  
75%           30.000000  
max          118.000000

In [63]:
# como se comportan las variables categóricas-- count=cantidad, unique=registros unicos, top=mas veces registrado, freq=frecuencia de los registros con mas aparariociones
Products.describe(include=['O'])

product_id product_category_name
count                              32951                 32341
unique                             32951                    73
top     1e9e8ef04dbcff4541ed26657ea517e5       cama_mesa_banho
freq                                   1                  3029

In [64]:
# cuantos datos nulos por columna se encuentran
Products.isna().sum()

product_id                      0
product_category_name         610
product_name_lenght           610
product_description_lenght    610
product_photos_qty            610
product_weight_g                2
product_length_cm               2
product_height_cm               2
product_width_cm                2
dtype: int64

## Sellers

In [65]:
# carga del dataset y muestreo de 10 registros aleatorios
Sellers = pd.read_csv('Datasets/olist_sellers_dataset.csv')
Sellers.sample(10)

seller_id  seller_zip_code_prefix  \
2452  e627629ba868740e287800f1a9be81c2                    1421   
2602  d23019c84ffae2d5ef2270367b8605fc                   86073   
834   916748bc99315c2d202898ae58b1617e                   18040   
2679  6861de0f40cacb0602e473c51639f980                   80420   
2352  76de363de16dacc923eec8daf91df504                   89825   
1220  2528744c5ef5d955adc318720a94d2e7                   13202   
2968  c840d3fdbba0790404fdae73d253b253                   81530   
784   dc120d932ddf9d4dfb6fd68beefda622                   83010   
2396  cad10cc982ab6a391570c211b995c17c                   18701   
1262  48985c61529077fa4f1e38bcff0f2ed3                   90020   

               seller_city seller_state  
2452             sao paulo           SP  
2602              londrina           PR  
834               sorocaba           SP  
2679              curitiba           PR  
2352                 xaxim           SC  
1220               jundiai           SP  
2968              curitiba           PR  
784   sao jose dos pinhais           PR  
2396                 avare           SP  
1262          porto alegre           RS

In [66]:
# mombres de columnas
Sellers.columns

Index(['seller_id', 'seller_zip_code_prefix', 'seller_city', 'seller_state'], dtype='object')

In [67]:
# tamaño del dataset
Sellers.shape

(3095, 4)

In [68]:
# tipo de datos, no nulos y cantidad de registros para cada columna
Sellers.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3095 entries, 0 to 3094
Data columns (total 4 columns):
 #   Column                  Non-Null Count  Dtype 
---  ------                  --------------  ----- 
 0   seller_id               3095 non-null   object
 1   seller_zip_code_prefix  3095 non-null   int64 
 2   seller_city             3095 non-null   object
 3   seller_state            3095 non-null   object
dtypes: int64(1), object(3)
memory usage: 96.8+ KB


In [69]:
# descripcion de las columnas numéricas
Sellers.describe()

seller_zip_code_prefix
count             3095.000000
mean             32291.059451
std              32713.453830
min               1001.000000
25%               7093.500000
50%              14940.000000
75%              64552.500000
max              99730.000000

In [70]:
# como se comportan las variables categóricas-- count=cantidad, unique=registros unicos, top=mas veces registrado, freq=frecuencia de los registros con mas aparariociones
Sellers.describe(include=['O'])

seller_id seller_city seller_state
count                               3095        3095         3095
unique                              3095         611           23
top     3442f8959a84dea7ee197c632cb2df15   sao paulo           SP
freq                                   1         694         1849

In [71]:
# cuantos datos nulos por columna se encuentran
Sellers.isna().sum()

seller_id                 0
seller_zip_code_prefix    0
seller_city               0
seller_state              0
dtype: int64

## Product_category_name_translation

In [72]:
# carga del dataset y muestreo de 10 registros aleatorios
Product_category_name_translation = pd.read_csv('Datasets/product_category_name_translation.csv')
Product_category_name_translation.sample(10)

product_category_name product_category_name_english
39           livros_tecnicos               books_technical
24              climatizacao              air_conditioning
21          fashion_calcados                 fashion_shoes
10         alimentos_bebidas                    food_drink
40           casa_construcao             home_construction
20                     audio                         audio
13  tablets_impressao_imagem        tablets_printing_image
57         livros_importados                books_imported
58                   bebidas                        drinks
42               moveis_sala         furniture_living_room

In [73]:
# mombres de columnas
Product_category_name_translation.columns

Index(['product_category_name', 'product_category_name_english'], dtype='object')

In [74]:
# tamaño del dataset
Product_category_name_translation.shape

(71, 2)

In [75]:
# tipo de datos, no nulos y cantidad de registros para cada columna
Product_category_name_translation.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 71 entries, 0 to 70
Data columns (total 2 columns):
 #   Column                         Non-Null Count  Dtype 
---  ------                         --------------  ----- 
 0   product_category_name          71 non-null     object
 1   product_category_name_english  71 non-null     object
dtypes: object(2)
memory usage: 1.2+ KB


In [76]:
# descripcion de las columnas numéricas
Product_category_name_translation.describe()

product_category_name product_category_name_english
count                     71                            71
unique                    71                            71
top             beleza_saude                 health_beauty
freq                       1                             1

In [77]:
# como se comportan las variables categóricas-- count=cantidad, unique=registros unicos, top=mas veces registrado, freq=frecuencia de los registros con mas aparariociones
Product_category_name_translation.describe(include=['O'])

product_category_name product_category_name_english
count                     71                            71
unique                    71                            71
top             beleza_saude                 health_beauty
freq                       1                             1

In [78]:
# cuantos datos nulos por columna se encuentran
Product_category_name_translation.isna().sum()

product_category_name            0
product_category_name_english    0
dtype: int64

# Carga en MySQL

In [79]:
from sqlalchemy.engine import create_engine
from urllib.parse import quote_plus as urlquote
from Pass_SQL import contraseña

In [80]:
str_conexion = 'mysql+pymysql://root:%s@localhost:3306/Proyecto_Grupal' % urlquote(contraseña())
conexion = create_engine(str_conexion)

In [81]:
# Cargamos los dataset en una tabla
Product_category_name_translation.to_sql(name='product_category_name_translation', con=conexion)
Sellers.to_sql(name='sellers', con=conexion)
Products.to_sql(name='products', con=conexion)
Orders.to_sql(name='orders', con=conexion)
Order_reviews.to_sql(name='order_reviews', con=conexion)
Order_payments.to_sql(name='order_payments', con=conexion)
Order_items.to_sql(name='order_items', con=conexion)
Marketing.to_sql(name='marketing', con=conexion)
Geolocation.to_sql(name='geolocation', con=conexion)
Customers.to_sql(name='customers', con=conexion)
Closed_deals.to_sql(name='closed_deals', con=conexion)

c:\Users\eugen\AppData\Local\Programs\Python\Python310\lib\site-packages\pandas\io\sql.py:1666: UserWarning: The provided table name 'Product_category_name_translation' is not found exactly as such in the database after writing the table, possibly due to case sensitivity issues. Consider using lower case table names.
  warnings.warn(msg, UserWarning)
c:\Users\eugen\AppData\Local\Programs\Python\Python310\lib\site-packages\pandas\io\sql.py:1666: UserWarning: The provided table name 'Sellers' is not found exactly as such in the database after writing the table, possibly due to case sensitivity issues. Consider using lower case table names.
  warnings.warn(msg, UserWarning)
c:\Users\eugen\AppData\Local\Programs\Python\Python310\lib\site-packages\pandas\io\sql.py:1666: UserWarning: The provided table name 'Products' is not found exactly as such in the database after writing the table, possibly due to case sensitivity issues. Consider using lower case table names.
  warnings.warn(msg, UserW

842